packages/next-server/lib/router/router.ts

```diff
@@ -579,13 +579,29 @@ export default class Router implements BaseRouter {
     }
     this.clc = cancel
     const { Component: App } = this.components['/_app']
+    let props
 
-    const props = await loadGetInitialProps<AppContextType<Router>>(App, {
-      AppTree: this._wrapApp(App),
-      Component,
-      router: this,
-      ctx,
-    })
+    if ((Component as any).__NEXT_PRERENDER) {
+      const url = format({
+        pathname: ctx.asPath,
+        query: ctx.query,
+      })
+      const res = await fetch(url, {
+        headers: { 'content-type': 'application/json' },
+      })
+      props = {
+        pageProps: res.ok
+          ? await res.json()
```

Comment:
We should handle this potential JSON parsing error, IMO.

Reply:
Updated